schemas/interface.ts

```diff
@@ -1,18 +1,186 @@
+         */
+        explanation: string;
+        /**
+         * Feedback when answer is wrong.
+         */
+        when_wrong?: string;
+        /**
+         * Feedback when answer is right.
+         */
+        when_right?: string;
+    }
+
+    /** 
+     * Question difficulty: easy, medium or hard.
+     */
+    type Difficulty = "EASY" | "MEDIUM" | "HARD";
+
+    /**
+     * Question spec. It depends on the question kind.
+     */
+    type QuestionSpec = ChoiceQuestion | TextQuestion | UploadQuestion | CodeQuestion;
+
+    /**
+     * Choice question : can either be a [Single Choice Question]{@link SingleChoiceQuestion} or a [Multiple Choices Question]{@link MultipleChoicesQuestion}.
      */
-    id: string;
+    type ChoiceQuestion = SingleChoiceQuestion | MultipleChoicesQuestion;
+
+    interface SingleChoiceQuestion {
+        /**
+         * Question options. See {@link QuestionOption}.
+         */
+        options: QuestionOption[]
+        /**
+         * Id of the correct answer to the question.
+         */
+        answer_id: OptionId;
+    }
+
+    interface MultipleChoicesQuestion {
+        /**
+         * Question options. See {@link QuestionOption}.
+         */
+        options: QuestionOption[]
+        /**
+         * Id of the correct answers to the question.
+         */
+        answers_id: OptionId[];
+    }
+
+    interface QuestionOption {
+        /**
+         * Id of the option.
+         */
+        id: OptionId;
+        /**
+         * The actual display of the option as a human readable string.
+         */
+        display: string;
+    }
+
+    type OptionId = string;
+
+    interface TextQuestion {
+        /**
+         * List of keywords that the answers should contain. It is up to the developer to implement if the answer should contain ALL or ANY of the keywords.
+         * 
+         * @examples ["bytes", "octets"]
+         */
+        keywords?: string[];
+    }
+
+    interface UploadQuestion {
+        /**
+         * List of files to upload.
+         */
+        files: File[];
+        /**
+         * Maximum size of each files. Can be expressed in bytes, megabytes (MB), etc. See TODO for valid units.
+         * 
+         * @examples ["10MB"]
+         */
+        max_size: string;
+        /** 
+         * Maximum number of files to upload.
+         * 
+         * @default 1
+         * @minimum 1
+         */
+        max_files: number;
+    }
+
+    interface File {
+        /**
+         * File type enum. Can be `.txt`, `.csv`, `.pdf`, etc. See {@link FileType}.
+         */
+        type: FileType;
+        /**
+         * Encoded `base64` string of the file's content.
+         */
+        content: string;
+    }
+
     /**
-     * The actual question as a human readable string.
+     * Enum of supported file extensions.
      */
-    question: string;
+    type FileType = ".txt" | ".csv" | ".pdf" | ".img" | ".jpg" | ".jpeg" | ".png";
+
+    type CodeQuestion = {
```

Comment:
You defined `ChoiceQuestion`, `UploadQuestion` and `TextQuestion` as interfaces, but `CodeQuestion` as a type ? 
Probably leading to [this](https://github.com/YacineSteeve/quizoot/pull/27/files#r1029766789)

Reply:
Oh my bad. Ty. I'll change that